### ISSUE TITLE
SSL exception when sending commands

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#284]
```
Heya,

SSL doesn't seem to work right for me. As soon as I send a command or two, 
gtalksms seems to disconnect.

As the XMPP server, I use Prosody 0.8.2, with a selfsigned certificate

Relevant parts of the log:

GTalkSMS Preferences
notifiedAddress: me@the-compiler.org
useGoogleMapUrl: true
locale: default
smsMagicWord: GTalkSMS
manuallySpecifyServerSettings: true
formatResponses: false
framebufferMode: ARGB_8888
batteryNotificationInterval: 1
useOpenStreetMapUrl: true
serviceName: the-compiler.org
startAtBoot: false
showStatusIcon: true
forceMucServer: true
markSmsReadOnReply: true
notificationIncomingSmsType: both
notifyBatteryInStatus: true
useCompression: false
login: phone
displayIconIndex: 0
smsReplySeparate: true
notifySmsDelivered: true
xmppSecurityMode: req
debugLog: true
smsNumber: 20
connectOnMainscreenShow: true
notifySmsSent: true
callLogsNumber: 10
publicIntentTokenRequired: true
roomPassword: gtalksms
serverHost: the-compiler.org
showSentSms: true
serverPort: 5222
mucServer: conference.the-compiler.org
notifyApplicationConnection: true
notifyBattery: true
stopOnPowerDelay: 1
notifyIncomingCalls: true
publicIntentToken: secret

GTalkSMS Version: 4.1
Android API: 15
Kernel info: 3.0.32-CM-ge9cef6c
pinky-ghost@cyanogenmod )
#1 SMP PREEMPT Sun Aug 5 01:41:29 PDT 2012


[...]

08-07 09:40:45.893 I/gtalksms(15148): Sending message "Available commands: - 
"help": shows..." to me@the-compiler.org/ws018
08-07 09:40:45.973 I/gtalksms(15148): onStartCommand(): Intent 
com.googlecode.gtalksms.action.XMPP.CONNECTION_CHANGED
08-07 09:40:45.973 I/gtalksms(15148): TTS initialized!
08-07 09:40:47.078 I/gtalksms(15148): handled action 
'com.googlecode.gtalksms.action.XMPP.MESSAGE_RECEIVED' - state now: Connected
08-07 09:40:47.078 I/gtalksms(15148): handling action 
'com.googlecode.gtalksms.action.SEND' while in state Connected
08-07 09:40:47.083 I/gtalksms(15148): Sending message "Welcome to GTalkSMS 4.1. 
Send "?" f..."
08-07 09:40:47.093 I/gtalksms(15148): handled action 
'com.googlecode.gtalksms.action.SEND' - state now: Connected
08-07 09:40:47.093 I/gtalksms(15148): handling action 
'com.googlecode.gtalksms.action.XMPP.CONNECTION_CHANGED' while in state 
Connected
08-07 09:40:47.093 I/gtalksms(15148): handled action 
'com.googlecode.gtalksms.action.XMPP.CONNECTION_CHANGED' - state now: Connected
08-07 09:41:57.545 W/gtalksms(15148): xmpp disconnected due to error:
08-07 09:41:57.545 W/gtalksms(15148): javax.net.ssl.SSLException: Read error: 
ssl=0x23bbc20: I/O error during system call, Connection reset by peer
08-07 09:41:57.545 W/gtalksms(15148): at 
org.apache.harmony.xnet.provider.jsse.NativeCrypto.SSL_read(Native Method)
08-07 09:41:57.545 W/gtalksms(15148): at 
org.apache.harmony.xnet.provider.jsse.OpenSSLSocketImpl$SSLInputStream.read(Open
SSLSocketImpl.java:671)
08-07 09:41:57.545 W/gtalksms(15148): at 
java.io.InputStreamReader.read(InputStreamReader.java:244)
08-07 09:41:57.545 W/gtalksms(15148): at 
java.io.BufferedReader.read(BufferedReader.java:310)
08-07 09:41:57.545 W/gtalksms(15148): at 
org.kxml2.io.KXmlParser.fillBuffer(KXmlParser.java:1496)
08-07 09:41:57.545 W/gtalksms(15148): at 
org.kxml2.io.KXmlParser.peekType(KXmlParser.java:979)
08-07 09:41:57.545 W/gtalksms(15148): at 
org.kxml2.io.KXmlParser.next(KXmlParser.java:346)
08-07 09:41:57.545 W/gtalksms(15148): at 
org.kxml2.io.KXmlParser.next(KXmlParser.java:310)
08-07 09:41:57.545 W/gtalksms(15148): at 
org.jivesoftware.smack.PacketReader.parsePackets(PacketReader.java:320)
08-07 09:41:57.545 W/gtalksms(15148): at 
org.jivesoftware.smack.PacketReader.access$000(PacketReader.java:43)
08-07 09:41:57.545 W/gtalksms(15148): at 
org.jivesoftware.smack.PacketReader$1.run(PacketReader.java:69)
08-07 09:41:57.860 I/gtalksms(15148): broadcasting state transition from 
Connected to Waiting to connect via Intent 
com.googlecode.gtalksms.action.XMPP.CONNECTION_CHANGED
08-07 09:41:57.885 I/gtalksms(15148): maybeStartReconnect scheduling retry in 
5000ms. Retry #1
08-07 09:41:57.890 I/gtalksms(15148): onStartCommand(): Intent 
com.googlecode.gtalksms.action.XMPP.CONNECTION_CHANGED
08-07 09:41:57.890 I/gtalksms(15148): handling action 
'com.googlecode.gtalksms.action.XMPP.CONNECTION_CHANGED' while in state Waiting 
to connect
08-07 09:41:57.890 I/gtalksms(15148): handled action 
'com.googlecode.gtalksms.action.XMPP.CONNECTION_CHANGED' - state now: Waiting 
to connect
08-07 09:46:35.143 I/gtalksms(15148): Sending message "Battery level 66%"
08-07 09:46:35.148 D/gtalksms(15148): Adding message: "Battery level 66%" to 
offline queue, because we are not connected. Status=Waiting to connect
08-07 09:46:35.148 I/gtalksms(15148): xmppRequestStateChange Waiting to connect 
=> Waiting to connect
```

Original issue reported on code.google.com by `m...@the-compiler.org` on 7 Aug 2012 at 8:00

**Comment by GoogleCodeExporter** [comment]
```
Issue 344 has been merged into this issue.
```

Original comment by `Florent....@gmail.com` on 21 Dec 2013 at 10:14

**Comment by GoogleCodeExporter** [comment]
```
Probably fixed with the last version.
Please reopen if still present
```

Original comment by `Florent....@gmail.com` on 17 Aug 2014 at 2:34
- Changed state: **FixedInBeta**

**Comment by GoogleCodeExporter** [comment]
```
Released on Play Market as 5.0
```

Original comment by `Florent....@gmail.com` on 25 Oct 2014 at 6:23
- Changed state: **Deployed**
